### ISSUE TITLE
S3 error - access denied

### ISSUE THREAD

**Issue by DBHertel** [aws-quickstart/quickstart-bitnami-wordpress#7]
I've repeatedly tried to use this new template, but can't get past the S3 error (below).  I'm an AWS administrator and have also ensured my AWS account has attached policy:  **AmazonS3FullAccess**

I can't find any more logging detail than what is shown below.  This error occurs within the first few seconds of launching the quickstart template.  My AWS environment meets all the requirements as stated, but it seems the S3 security isn't correct.  Can someone assist please?

CREATE_FAILED | AWS::CloudFormation::Stack | SecurityGroupsStack |
   S3 error: Access Denied For more information check http://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html

**Comment by gavinharriscolab** [comment]
I found an issue on line 559 of the wordpress-master.template.

The URL appears to be created as such:
`https://${QSS3BucketName}.s3.amazonaws.com/${QSS3KeyPrefix}submodules/quickstart-aws-vpc/templates/aws-vpc.template`

However when I plug in the default values this creates a URL as such:

https://aws-quickstart.s3.amazonaws.com/quickstart-bitnami-wordpresssubmodules/quickstart-aws-vpc/templates/aws-vpc.template

This fails with Access Denied. However if you change the URL to:

https://aws-quickstart.s3.amazonaws.com/quickstart-bitnami-wordpress/submodules/quickstart-aws-vpc/templates/aws-vpc.template

It works.

![image](https://user-images.githubusercontent.com/5170199/46120576-a6b8c080-c253-11e8-8e95-1c76464d24bc.png)

To work around I added a `/` to the 'Quick Start S3 Key Prefix', so it became:  `quickstart-bitnami-wordpress/`.

If I get a chance today I will create a PR with a fix.

**Comment by agamzn** [comment]
Thanks. If the PR isn’t submitted by morning (US / Pacific), I will push a commit to fix this. Typo on my part. Apologies.

Sent from my iPhone

On Sep 26, 2018, at 19:52, gavinharriscolab <notifications@github.com<mailto:notifications@github.com>> wrote:


I found an issue on line 559 of the wordpress-master.template.

The URL appears to be created as such:
https://${QSS3BucketName}.s3.amazonaws.com/${QSS3KeyPrefix}submodules/quickstart-aws-vpc/templates/aws-vpc.template<https://${qss3bucketname}.s3.amazonaws.com/${QSS3KeyPrefix}submodules/quickstart-aws-vpc/templates/aws-vpc.template>

However when I plug in the default values this creates a URL as such:

https://aws-quickstart.s3.amazonaws.com/quickstart-bitnami-wordpresssubmodules/quickstart-aws-vpc/templates/aws-vpc.template

This fails with Access Denied. However if you change the URL to:

https://aws-quickstart.s3.amazonaws.com/quickstart-bitnami-wordpress/submodules/quickstart-aws-vpc/templates/aws-vpc.template

It works.

[image]<https://user-images.githubusercontent.com/5170199/46120576-a6b8c080-c253-11e8-8e95-1c76464d24bc.png>

To work around I added a / to the 'Quick Start S3 Key Prefix', so it became: quickstart-bitnami-wordpress/.

If I get a chance today I will create a PR with a fix.

**Comment by gavinharriscolab** [comment]
Just a note, the Environment build perfectly by appending the `/`.

Thank you so much for putting this together!

**Comment by agamzn** [comment]
#8 has been merged into develop. Our CI/CD will merge to master after a successful build. Expect that in 60-90 minutes.

FYI, this wasn't caught because I included the extra slash in my CI params. 

https://github.com/aws-quickstart/quickstart-bitnami-wordpress/blob/119a2105be44814a1a07247dfb5210b3e56ec155/ci/wordpress-master-nossl.json#L8

The default value for the QSS3KeyPrefix param does not have that slash. I've updated the CI parameters in https://github.com/aws-quickstart/quickstart-bitnami-wordpress/commit/60fdf7df981bec6c19b99f676133b6556414dea0 . Thanks again for the contributions! I'll leave this open until everything's in master.

**Comment by DBHertel** [comment]
Update -- I just retried, adding the slash "/" to the end.. the error no longer occurred (Yeah!) . but unfortunately, new errors cropped up later.  While trying to create the RDS db

Embedded stack arn:aws:cloudformation:us-west-2:811635507994:stack/WordPress-HA-by-Bitnami-Volteo8-RDSAuroraStack-78R4DLMDPH18/b4a69fd0-c26e-11e8-8a56-062d9ae1b752 was not successfully created: Th**e following resource(s) failed to create: [AuroraDBSecondaryInstance, AuroraDBPrimaryInstance].**

**Comment by agamzn** [comment]
Sanity checks: Are you up against any resource limits?

**Comment by DBHertel** [comment]
I don't believe so...  we have only 1 other RDS db instance.  And as for EC2 machines we have < 5 other EC2 vm's in our AWS env. sandbox.

**Comment by DBHertel** [comment]
11 minutes in... and yet another failure :( . 


The following resource(s) failed to create: [**NotificationTopic, WordpressEFS, ApplicationLoadBalancer].**
--
  | Physical ID:arn:aws:cloudformation:us-west-2:811635507994:stack/WordPress-HA-by-Bitnami-Volteo-WebserverStack-17MXX0DKEHHP/98372050-c272-11e8-8214-0ae06257a2d8

**Comment by DBHertel** [comment]
Just discovered my 2 public subnets are in same zone.... hence the build error.   I'm working on adding a subnet to my VPC, in a different AZ and will try again when ready.   in mean time, if you wish to email you can reach me at DBHertel  at gmail

**Comment by DBHertel** [comment]
Build successful when A) I had to public subnets in 2 diff zones and B) appended a "/" to the end of the quickstart...

Thanks for help, building this CFT and quick replies too

**Comment by agamzn** [comment]
Quite welcome! Glad to be of assistance. The CI build that'll address the forward-slash is still running. I'll leave this open until that fix is confirmed to be in the master branch.
